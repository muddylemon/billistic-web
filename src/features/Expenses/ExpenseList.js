import React from "react"
import "./expense_list.scss"
import { compose } from "recompose"
import { withExpenseHandlers, withExpenseData } from './Expense.hocs'

import Table from '@material-ui/core/Table'
import TableBody from '@material-ui/core/TableBody'
import TableCell from '@material-ui/core/TableCell'
import TableHead from '@material-ui/core/TableHead'
import TableRow from '@material-ui/core/TableRow'
import Paper from '@material-ui/core/Paper'
import Checkbox from '@material-ui/core/Checkbox'

const Expenses = ({ expenses, togglePaid }) => expenses.map(expense => (
  <TableRow key={expense.id}>
    <TableCell>
      <Checkbox className="mark_paid" checked={expense.paid} onClick={() => togglePaid(expense.id, !expense.paid)} />
    </TableCell>
    <TableCell component="th" scope="row">{expense.name}</TableCell>
    <TableCell numeric>{expense.amount}</TableCell>
    <TableCell numeric>{expense.due}</TableCell>
  </TableRow>
))

export const billList = ({ expenses, togglePaid }) => (
  <Paper>
    <Table>
      <TableHead>
        <TableRow>
          <TableCell>paid</TableCell>
          <TableCell>Biller</TableCell>
          <TableCell numeric>Amount</TableCell>
          <TableCell numeric>Due</TableCell>
        </TableRow>
      </TableHead>
      <TableBody>
        {!expenses ? (<TableRow><TableCell>Not loaded</TableCell></TableRow>) : (<Expenses expenses={expenses} togglePaid={togglePaid} />)}
      </TableBody>
    </Table>
  </Paper>
)

export const BillList = compose(
  withExpenseData,
  withExpenseHandlers
)(billList)
