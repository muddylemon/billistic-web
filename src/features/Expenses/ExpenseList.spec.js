import React from "react"
import { BillList } from "./ExpenseList"
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'

describe('BillList', () => {
  const initialState = {
    firestore: {
      ordered: {
        expenses: [
          {
            name: 'bank',
            paid: false,
            amount: 20,
            due: 20,
            id: 321,
          },
          {
            name: 'not bank',
            paid: true,
            amount: 20000,
            due: 1,
            id: 123,
          },
        ],
      },
    },
  }

  let store

  beforeEach(() => {
    const mockStore = configureStore()
    store = mockStore(initialState)
  })

  it('should render a new line for each expense', () => {
    const tree = renderer.create(<Provider store={store}><BillList /></Provider>)
    expect(tree).toMatchSnapshot()
  })
})
