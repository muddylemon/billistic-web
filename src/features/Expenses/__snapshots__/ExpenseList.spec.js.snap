// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BillList should render a new line for each expense 1`] = `
<div
  className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 expense_list"
>
  <h2
    className="MuiTypography-root-28 MuiTypography-h5-44 expense_list-title"
  >
    Expenses
  </h2>
  <table
    className="MuiTable-root-63"
  >
    <thead
      className="MuiTableHead-root-64"
    >
      <tr
        className="MuiTableRow-root-65 MuiTableRow-head-68"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root-70 MuiTableCell-head-71"
          scope="col"
        >
          paid
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-70 MuiTableCell-head-71"
          scope="col"
        >
          Biller
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-70 MuiTableCell-head-71 MuiTableCell-numeric-74"
          scope="col"
        >
          Amount
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-70 MuiTableCell-head-71 MuiTableCell-numeric-74"
          scope="col"
        >
          Due
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root-78"
    >
      <tr
        className="MuiTableRow-root-65"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-70 MuiTableCell-body-72"
        >
          <span
            className="MuiButtonBase-root-95 MuiIconButton-root-89 MuiPrivateSwitchBase-root-85 MuiCheckbox-root-79 MuiCheckbox-colorSecondary-84 mark_paid"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label-94"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-98"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
              <input
                checked={false}
                className="MuiPrivateSwitchBase-input-88"
                data-indeterminate={false}
                onChange={[Function]}
                type="checkbox"
              />
            </span>
            <span
              className="MuiTouchRipple-root-107"
            />
          </span>
        </td>
        <th
          aria-sort={null}
          className="MuiTableCell-root-70 MuiTableCell-body-72"
          scope="row"
        >
          bank
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root-70 MuiTableCell-body-72 MuiTableCell-numeric-74"
        >
          20
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-70 MuiTableCell-body-72 MuiTableCell-numeric-74"
        >
          20
        </td>
      </tr>
      <tr
        className="MuiTableRow-root-65"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-70 MuiTableCell-body-72"
        >
          <span
            className="MuiButtonBase-root-95 MuiIconButton-root-89 MuiPrivateSwitchBase-root-85 MuiCheckbox-root-79 MuiCheckbox-colorSecondary-84 MuiPrivateSwitchBase-checked-86 MuiCheckbox-checked-80 mark_paid"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label-94"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-98"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
              <input
                checked={true}
                className="MuiPrivateSwitchBase-input-88"
                data-indeterminate={false}
                onChange={[Function]}
                type="checkbox"
              />
            </span>
            <span
              className="MuiTouchRipple-root-107"
            />
          </span>
        </td>
        <th
          aria-sort={null}
          className="MuiTableCell-root-70 MuiTableCell-body-72"
          scope="row"
        >
          not bank
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root-70 MuiTableCell-body-72 MuiTableCell-numeric-74"
        >
          20000
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-70 MuiTableCell-body-72 MuiTableCell-numeric-74"
        >
          1
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
